test/integration/connection-monitoring-and-pooling/connection.test.js

```diff
@@ -452,6 +452,26 @@ describe('Connection', function () {
       })
     );
 
+    it('throws when attempting an operation if the client is not connected', function (done) {
+      const client = this.configuration.newClient();
+      const collection = client.db('shouldCorrectlyFailOnRetry').collection('test');
+      collection.insertOne({ a: 2 }, err => {
+        expect(err).to.match(/must be connected/);
```

Comment:
Can we check for instanceof `MongoNotConnectedError` class instead, error messages are subject to change.